lib/Auth/Source/scoutnetauth.php

```diff
@@ -109,4 +108,126 @@ protected function login($username, $password)
             throw new SimpleSAML_Error_Error('WRONGUSERPASS');
         }
     }
+
+    /**
+     * Formats a name to be used in an email address.
+     * @param string $name
+     * @return string
+     */
+    private function formatNameForEmail($name)
+    {
+        $name = trim(strtolower($name)); // Convert to lower case and trim spaces.
+        $name = preg_replace('/\s+/', '.', $name); // Replace empty space with one dot each.
+        $name = preg_replace('/[.][\-]/', '-', $name); // Replace .- with -
+        $name = preg_replace('/[\-][.]/', '-', $name); // Replace -. with -
+        $name = $this->removeDiacritics($name); // Replace special characters with normal ones.
```

Comment:
strtolower should be done after reomveDiacritics, or name like "Åsa" becomes "Asa" insted of "asa".
ref: https://3v4l.org/SLcV7

Reply:
For reference:
The javascript-version can do .toLowerCase() first, as it converts "Å"->"å", phps strtolower() dosn't convert "Å".